osu.Game/Screens/SelectV2/FooterButtonOptions_Popover.cs

```diff
@@ -60,32 +59,34 @@ private void load(ManageCollectionsDialog? manageCollectionsDialog, OsuColour co
                     Spacing = new Vector2(3),
                 };
 
-                beatmapWhenOpening = beatmap.Value;
-
                 addHeader(CommonStrings.General);
                 addButton(SongSelectStrings.ManageCollections, FontAwesome.Solid.Book, () => manageCollectionsDialog?.Show());
 
-                addHeader(SongSelectStrings.ForAllDifficulties, beatmapWhenOpening.BeatmapSetInfo.ToString());
-                addButton(SongSelectStrings.DeleteBeatmap, FontAwesome.Solid.Trash, () => { }, colours.Red1); // songSelect?.DeleteBeatmap(beatmapWhenOpening.BeatmapSetInfo);
+                addHeader(SongSelectStrings.ForAllDifficulties, beatmap.BeatmapSetInfo.ToString());
+                addButton(SongSelectStrings.DeleteBeatmap, FontAwesome.Solid.Trash, () => songSelect.RequestDeleteBeatmap(beatmap.BeatmapSetInfo), colours.Red1);
```

Comment:
why are we not using DI for this and passing song select down multiple levels of components?

Reply:
This button exists outside of `SongSelect`'s hierarchy, it's part of the footer, and the footer is not part of song select. The best method to have this here is to pass it down from `SongSelect` to the button to the popover.